React-Template-master/src/pages/Register.tsx

```diff
@@ -1,6 +1,180 @@
-import React from 'react';
-import { Container } from '@mui/material';
+import AppLog from '../assets/Lejit.svg';
+import React, { useEffect, useState } from 'react';
+import { Button, TextField, Link, Box, Grid, Typography } from '@mui/material';
+import { PasswordField } from '../components/PasswordField';
 
-export function Register(): React.ReactElement {
-    return <Container>Register</Container>;
+interface SignUpInputObject {
+    firstName: string;
+    lastName: string;
+    email: string;
+    password: string;
 }
+
+const checkForEmptyInputs = (data: SignUpInputObject): boolean => {
+    const allValues = Object.values(data);
+    for (const value of allValues) {
+        if (value === '') return true;
+    }
+    return false;
+};
+
+export const Register = (): JSX.Element => {
+    const handleSubmit = (event: React.FormEvent<HTMLFormElement>): void => {
+        event.preventDefault();
+        const data = new FormData(event.currentTarget);
+        console.log({
+            email: data.get('email'),
+            password: data.get('password')
+        });
+    };
+
+    const [data, setData] = useState({
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: ''
+    });
+
+    const setPassword = (password: string): void => {
+        setData({ ...data, password });
+    };
+
+    const [btnColor, setBtnColor] = useState('#EBEBF0');
```

Comment:
Use meaningful names and store and use colors from palette

Reply:
okay